src/uart.rs

```diff
@@ -189,6 +189,85 @@ pub mod config {
         }
     }
 
+    /// UART clock source
+    #[derive(PartialEq, Eq, Copy, Clone, Debug)]
+    pub enum SourceClock {
+        /// UART source clock from `APB`
+        APB,
+        /// UART source clock from `RTC`
+        #[cfg(esp_idf_soc_uart_support_rtc_clk)]
+        RTC,
+        /// UART source clock from `XTAL`
+        #[cfg(esp_idf_soc_uart_support_xtal_clk)]
+        Crystal,
+        /// UART source clock from `REF_TICK`
+        #[cfg(esp_idf_soc_uart_support_ref_tick)]
+        RefTick,
+    }
+
+    impl Default for SourceClock {
+        fn default() -> Self {
+            #[cfg(not(esp_idf_version_major = "4"))]
+            let source_clock = soc_periph_uart_clk_src_legacy_t_UART_SCLK_DEFAULT;
```

Comment:
In `espidf v5` the `uart_sclk_t` no longer starts from a default 0, but the default is `APB` for all chips, so this function can just be replace with a generic `derive(Default)`. this is however explicit and forwards compatible (hopefully). I started with the derive implementation, but I'm afraid about it somehow breaking along the way.